service-base/src/main/java/org/eclipse/hono/service/AbstractServiceBase.java

```diff
@@ -33,6 +35,14 @@
 
     private HttpServer healthCheckServer;
 
+    private ApplicationConfigProperties applicationConfig;
```

Comment:
What about changing Autowiring on the setter to `required = true` instead? Would also solve the problem with inconsistent defaults that I mentioned above.